src/main/java/edu/ucla/library/prl/harvester/handlers/UpdateJobHandler.java

```diff
@@ -47,4 +63,61 @@ public void handle(final RoutingContext aContext) {
         }
     }
+
+    /**
+     * Removes all item records that belong to the sets removed from the job (if any).
+     *
+     * @param aData A Tuple of the old JSON representation of the job, its new JSON representation, and the institution
+     *        name
+     */
+    @Override
+    Future<UpdateResponse> updateSolr(final Tuple aData) {
+        final Job oldJob = new Job(aData.getJsonObject(0));
+        final Job newJob = new Job(aData.getJsonObject(1));
+        final String institutionName = aData.getString(2);
+
+        final Optional<List<String>> oldJobSets = oldJob.getSets();
+        final Optional<List<String>> newJobSets = newJob.getSets();
+        final Future<List<String>> getActualOldJobSets;
+
+        if (oldJobSets.isPresent() && newJobSets.isPresent()) {
+            // Still selective harvesting, althought the list of sets may have changed, so determine that
+            getActualOldJobSets = Future.succeededFuture(oldJobSets.get());
+        } else if (oldJobSets.isEmpty() && newJobSets.isPresent()) {
+            // From non-selective harvesting to selective, so it's very likely that there are sets to remove
+            // Must query OAI-PMH repository in order to get the sets belonging to the old job
+            getActualOldJobSets =
+                    OaipmhUtils.listSets(myVertx, oldJob.getRepositoryBaseURL()).map(OaipmhUtils::getSetSpecs);
+            // TODO: make it impossible to change the base URL
+        } else if (oldJobSets.isPresent() && newJobSets.isEmpty()) {
+            // From selective harvesting to non-selective, so nothing to remove
+            return Future.succeededFuture();
+        } else {
+            // Still harvesting entire repository, so nothing to remove
+            return Future.succeededFuture();
+        }
+
+        // If we determined that there are sets to remove, do that now
+        return getActualOldJobSets.compose(sets -> {
+            final Optional<List<String>> setsToRemove = Optional.of(getElementsInNewNotInOld(sets, newJobSets.get()));
+            final Future<String> getSolrQuery = RemoveJobHandler.getRecordRemovalQuery(myVertx, institutionName,
+                    oldJob.getRepositoryBaseURL(), setsToRemove);
+
+            return getSolrQuery.compose(solrQuery -> {
+                final CompletionStage<UpdateResponse> removal =
+                        mySolrClient.deleteByQuery(solrQuery).thenCompose(result -> mySolrClient.commit());
+
+                return Future.fromCompletionStage(removal);
+            });
+        });
+    }
+
+    /**
+     * @param anOldList A list representing an original state
+     * @param aNewList A list representing a new, updated state
+     * @return The list of elements that are in {@code aNewList} but not in {@code anOldList}
+     */
+    private static List<String> getElementsInNewNotInOld(final List<String> anOldList, final List<String> aNewList) {
```

Comment:
`getDifference()`? The javadocs can explain the details instead of the method name.